### PR TITLE
Allow setting thermal conductivity components as function of temperature

### PR DESCRIPTION
**Pull request by singhgp4321**
Ref: #22027

## Reason
User may want to set individual components of thermal conductivity tensor as different functions of temperature.

## Design
The current model AnisoHeatConductionMaterial will be updated to allow user to input individual components. The user will set the components using input parameters which will be functions of temperature; functions will be defined in the input file.

## Impact
This update will not change the existing functionality of the code.

### CODE REVIEW

**Comment by dschwen** [comment]
A better Dr resign would be to have a material like GenericFunctionMaterial that builds a rank two tensor from functions. That way not every class would have to be modified to support function dependent tensors.

File: modules/heat_conduction/doc/content/source/materials/AnisoHeatConductionMaterial.md [hunk 1]
```diff
@@ -4,30 +4,37 @@
...
 
 \begin{equation}
  k(T) = k_{o} (1 + \alpha (T - T_{ref}) )
 \end{equation}
 
-where $k_{o}$ is the thermal conductivity based on the user supplied vector. The specific heat capacity $C_p$ can be supplied either as a constant value or as a function using the parameter `specific_heat`.
+where $k_{o}$ is the thermal conductivity based on the user supplied vector.
+
+
+Another way, by which the user can provide thermal conductivity, is by providing the indivifual components `k_11`, `k_22` and `k_33` which can be either a constant value or a function of temperature.  Note that if `k_22` or `k_33` are not provided then their values will be set equal to that of `k_11`. Of the three parameters `k_11`, `k_22` and `k_33`, `k_11` is mandatory and the latter two are optional.
```
Comment (dschwen) [inline modules/heat_conduction/doc/content/source/materials/AnisoHeatConductionMaterial.md:26]:
```suggestion
Another way, by which the user can provide thermal conductivity, is by providing the individual components `k_11`, `k_22` and `k_33` which can be either a constant value or a function of temperature.  Note that if `k_22` or `k_33` are not provided then their values will be set equal to that of `k_11`. Of the three parameters `k_11`, `k_22` and `k_33`, `k_11` is mandatory and the latter two are optional.
```

File: modules/heat_conduction/src/materials/AnisoHeatConductionMaterial.C [hunk 2]
```diff
@@ -28,12 +28,15 @@ AnisoHeatConductionMaterialTempl<is_ad>::validParams()
   params.addParam<Real>(
       "reference_temperature", 293.0, "Reference temperature for thermal conductivity in Kelvin.");
   params.addParam<std::string>("base_name", "Material property base name.");
-  params.addRequiredParam<std::vector<Real>>("thermal_conductivity",
-                                             "The thermal conductivity tensor values");
+  params.addParam<std::vector<Real>>("thermal_conductivity",
```
Comment (dschwen) [inline modules/heat_conduction/src/materials/AnisoHeatConductionMaterial.C:31]: Numbers are a special case of functions. Those should not be different parameters. In fact the class should couple a material property instead.

**Review by dschwen — CHANGES_REQUESTED** [review]
Factor out function tensor material

**Comment by singhgp4321** [comment]
@dschwen Could you please reopen this PR? I don't seem to have the rights to open it.

**Pull request closed without merging**